code-scanning/codacy.yml

```diff
@@ -22,8 +22,14 @@ on:
   schedule:
     - cron: $cron-weekly
 
+permissions:
```

Comment:
Why do we need to set `permissions` in two places (global and inside the `codacy-security-scan` job)?

Reply:
We've raised this same question on other PRs, and the rationale is that setting restrictive overall permissions is a defensive measure against future updates adding more jobs to the workflow and forgetting to explicitly set permissions for each individually.

So long as your happy that the permissions given to `codacy-security-scan` are appropriate then I think we're good to go! 🙂

Reply:
Yes, we tested the changes and the Codacy Action is working correctly.
You have our (I'm a Codacy employee) thumbs up for this change!
Thank you!

Reply:
> We've raised this same question on other PRs, and the rationale is that setting restrictive overall permissions is a defensive measure against future updates adding more jobs to the workflow and forgetting to explicitly set permissions for each individually.

It makes total sense. Thank you for the explanation.

Reply:
Great, thanks!